lib/actions/upsert_v2.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable no-param-reassign */
+const { messages } = require('elasticio-node');
+const { callJSForceMethod } = require('../helpers/wrapper');
+const { createProperty, getLookupFieldsModelWithTypeOfSearch } = require('../helpers/utils');
+const attachment = require('../helpers/attachment');
+
+module.exports.getObjectTypes = async function getObjectTypes(configuration) {
+  return callJSForceMethod.call(this, configuration, 'getObjectTypes');
+};
+
+module.exports.getLookupFieldsModel = async function getLookupFieldsModel(configuration) {
+  const meta = await callJSForceMethod.call(this, configuration, 'describe');
+  return getLookupFieldsModelWithTypeOfSearch(meta, configuration.typeOfSearch);
+};
+
+module.exports.getUpdateFieldsModel = async function getLookupFieldsModel(configuration) {
+  const meta = await callJSForceMethod.call(this, configuration, 'describe');
+  meta.fields = meta.fields.filter((field) => !field.deprecatedAndHidden
+    && field.updateable
+    && field.createable);
+  return getLookupFieldsModelWithTypeOfSearch(meta, '');
+};
+
+module.exports.getMetaModel = async function getMetaModel(configuration) {
+  const meta = await callJSForceMethod.call(this, configuration, 'describe');
+  const fields = meta.fields.filter((field) => !field.deprecatedAndHidden
+    && field.updateable
+    && field.createable
+    && (configuration.updateFields.includes(field.name) || (!field.nillable && !field.defaultedOnCreate)));
+
+  const result = {
+    in: {
+      type: 'object',
+      properties: {},
+    },
+    out: {
+      type: 'object',
+      properties: {
+        id: {
+          type: 'string',
+          required: true,
+          title: 'Unique identifier from CRM',
+        },
+        success: {
+          type: 'boolean',
+          required: true,
+          title: 'Request result',
+        },
+        errors: {
+          type: 'array',
+          required: true,
+          title: 'Founded errors',
+        },
+      },
+    },
+  };
+
+  result.in.properties[configuration.lookupField] = {
+    type: 'string',
+    title: `lookup by - ${configuration.lookupField}`,
+    required: configuration.lookupField !== 'Id',
+  };
+
+  fields.forEach((field) => {
+    result.in.properties[field.name] = createProperty(field);
+  });
+  return result;
+};
+
+module.exports.process = async function upsertObject(message, configuration) {
+  this.logger.info('Starting Upsert Object Action');
+  await attachment.prepareBinaryData(message, configuration, this);
+  let existingObj;
+  this.logger.info('Looking for Object');
+  if (configuration.lookupField === 'Id' && (message.body.Id === '' || !message.body.Id)) {
```

Comment:
`!message.body.Id` is true if `message.body.Id === ''`

Reply:
Absolutely right, removed.